### PR TITLE
Upgrade nelmio/alice

### PR DESCRIPTION
**Pull request by Neloop**

### CODE REVIEW

File: app/helpers/ZenifyFixtures/Alice/AliceLoader.php [hunk 1]
```diff
@@ -20,86 +20,65 @@
...
+        }
+        $this->entityManager->flush();
+
+        return $entities;
+    }
+
+    private function checkExistence(array $sources)
+    {
+        foreach ($sources as $source) {
+            if (!is_file($source)) {
```
Comment (krulis-martin) [inline app/helpers/ZenifyFixtures/Alice/AliceLoader.php:67]: Maybe we should also add is_readable() check? Just a thought.

File: app/helpers/ZenifyFixtures/Alice/CustomNativeLoader.php [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+    protected function getBlacklistedFunctions(): array
+    {
+        return [
+            'current',
+            'date'
+        ];
+    }
+}
```
Comment (krulis-martin) [inline app/helpers/ZenifyFixtures/Alice/CustomNativeLoader.php:42]: Yo, no newline!

File: app/helpers/ZenifyFixtures/Alice/Fixtures/Parser/Methods/NeonParser.php [hunk 3]
```diff
@@ -9,80 +9,53 @@
...
-				$includeFile = dirname($filename) . DIRECTORY_SEPARATOR . $include;
-				$includeData = $this->parse($includeFile);
-				$data = Helpers::merge($includeData, $data);
-			}
-			unset($data[$includeKeyword]);
-		}
-		return $data;
-	}
-
+    private const REGEX = '/.{1,}\.neon/i';
```
Comment (krulis-martin) [inline app/helpers/ZenifyFixtures/Alice/Fixtures/Parser/Methods/NeonParser.php:22]: Did you know, that `{1,}` can be written as `+` O:)

File: app/helpers/ZenifyFixtures/Alice/Fixtures/Parser/Methods/NeonParser.php [hunk 4]
```diff
@@ -9,80 +9,53 @@
...
+            throw InvalidArgumentExceptionFactory::createForFileCouldNotBeFound($file);
+        }
+
+        try {
+            $data = Neon::decode(file_get_contents($file));
+
+            if (null === $data) {
+                throw new UnparsableFileException(sprintf('The file "%s" does not contain valid NEON.', $file));
+            }
+
```
Comment (krulis-martin) [inline app/helpers/ZenifyFixtures/Alice/Fixtures/Parser/Methods/NeonParser.php:51]: `1 ===`, `null ===`, `false ===` ... why do I get the feeling that Yoda coded this...?

File: app/helpers/ZenifyFixtures/DI/FixturesExtension.php [hunk 5]
```diff
@@ -41,35 +39,19 @@ public function beforeCompile()
...
         $containerBuilder->resolve();
 
-        $this->loadFakerProvidersToAliceLoader();
-        $this->loadParsersToAliceLoader();
+        $this->loadFakerConfiguration();
     }
 
 
-    private function loadFakerProvidersToAliceLoader()
+    private function loadFakerConfiguration()
```
Comment (krulis-martin) [inline app/helpers/ZenifyFixtures/DI/FixturesExtension.php:46]: Faker?

File: app/helpers/ZenifyFixtures/Alice/AliceLoader.php (hunk 1 above)
Comment (Neloop) [reply app/helpers/ZenifyFixtures/Alice/AliceLoader.php:67]: yeah, why not

File: app/helpers/ZenifyFixtures/Alice/CustomNativeLoader.php (hunk 2 above)
Comment (Neloop) [reply app/helpers/ZenifyFixtures/Alice/CustomNativeLoader.php:42]: It got lost somewhere, I'll try to find it

File: app/helpers/ZenifyFixtures/Alice/Fixtures/Parser/Methods/NeonParser.php (hunk 3 above)
Comment (Neloop) [reply app/helpers/ZenifyFixtures/Alice/Fixtures/Parser/Methods/NeonParser.php:22]: Well, now I do

File: app/helpers/ZenifyFixtures/Alice/CustomNativeLoader.php (hunk 2 above)
Comment (krulis-martin) [reply app/helpers/ZenifyFixtures/Alice/CustomNativeLoader.php:42]: Look under the table!

File: app/helpers/ZenifyFixtures/Alice/Fixtures/Parser/Methods/NeonParser.php (hunk 4 above)
Comment (Neloop) [reply app/helpers/ZenifyFixtures/Alice/Fixtures/Parser/Methods/NeonParser.php:51]: Dunno

File: app/helpers/ZenifyFixtures/DI/FixturesExtension.php (hunk 5 above)
Comment (Neloop) [reply app/helpers/ZenifyFixtures/DI/FixturesExtension.php:46]: Faker is the name of the library which is used for data generation.
See https://github.com/FakerPHP/Faker

**Comment by Neloop** [comment]
Final polishing commit added

**Review by krulis-martin — APPROVED** [review]

**Pull request merged**